conjure-http/src/server.rs

```diff
@@ -468,61 +533,86 @@ impl ConjureResponseSerializer {
     }
 }
 
-impl<T, W> SerializeResponse<T, W> for ConjureResponseSerializer
+impl<'a, T, W> SerializeResponse<'a, T, W> for ConjureResponseSerializer<'a>
 where
     T: Serialize,
 {
+    #[inline]
+    fn new(runtime: &'a ConjureRuntime) -> Self {
+        ConjureResponseSerializer { _runtime: runtime }
+    }
+
     fn serialize(
+        &self,
         _request_headers: &HeaderMap,
         value: T,
     ) -> Result<Response<ResponseBody<W>>, Error> {
-        Self::serialize_inner(value, ResponseBody::Fixed)
+        self.serialize_inner(value, ResponseBody::Fixed)
     }
 }
 
-impl<T, W> AsyncSerializeResponse<T, W> for ConjureResponseSerializer
+impl<'a, T, W> AsyncSerializeResponse<'a, T, W> for ConjureResponseSerializer<'a>
 where
     T: Serialize,
 {
+    #[inline]
+    fn new(runtime: &'a ConjureRuntime) -> Self {
+        ConjureResponseSerializer { _runtime: runtime }
+    }
+
     fn serialize(
+        &self,
         _request_headers: &HeaderMap,
         value: T,
     ) -> Result<Response<AsyncResponseBody<W>>, Error> {
-        Self::serialize_inner(value, AsyncResponseBody::Fixed)
+        self.serialize_inner(value, AsyncResponseBody::Fixed)
     }
 }
 
 /// A trait implemented by header decoders used by custom Conjure server trait implementations.
-pub trait DecodeHeader<T> {
+pub trait DecodeHeader<'a, T> {
+    /// Creates a new decoder.
+    fn new(runtime: &'a ConjureRuntime) -> Self;
+
     /// Decodes the value from headers.
-    fn decode<'a, I>(headers: I) -> Result<T, Error>
+    fn decode<'b, I>(&self, headers: I) -> Result<T, Error>
     where
-        I: IntoIterator<Item = &'a HeaderValue>;
+        I: IntoIterator<Item = &'b HeaderValue>;
 }
 
 /// A trait implemented by URL parameter decoders used by custom Conjure server trait
 /// implementations.
-pub trait DecodeParam<T> {
+pub trait DecodeParam<'a, T> {
+    /// Creates a new decoder.
+    fn new(runtime: &'a ConjureRuntime) -> Self;
+
     /// Decodes the value from the sequence of values.
     ///
     /// The values have already been percent-decoded.
-    fn decode<I>(params: I) -> Result<T, Error>
+    fn decode<I>(&self, params: I) -> Result<T, Error>
     where
         I: IntoIterator,
         I::Item: AsRef<str>;
 }
 
 /// A decoder which converts a single value using its [`FromStr`] implementation.
-pub enum FromStrDecoder {}
+pub struct FromStrDecoder<'a> {
```

Comment:
Not sure if we'd ever want to pull dynamic logic for this kind of thing but I'm future proofing just in case.